### PR TITLE
Remote Host name resolution

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?

Adds a remote host name resolution component. 

### Motivation
We need to get hostname from core agent in OTel Agent. Moving remote host resolution to an independent module

### CODE REVIEW

File: comp/core/hostname/remotehostnameimpl/hostname.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
+	cache "github.com/patrickmn/go-cache"
+	"go.uber.org/fx"
+)
+
+const (
+	defaultExpire = 15 * time.Minute
```
Comment (truthbk) [inline comp/core/hostname/remotehostnameimpl/hostname.go:28]: Looks like these values are defaults but cannot be overridden, should these be configurable? Same goes for `maxAttempts`. That would mean, of course, we'd need a dependency with the configuration component.

**Review by truthbk — APPROVED** [review]
Looks good to me, but added a comment about the potential need to override certain defaults. If you don't think it's necessary, I'm OK with that, or if you'd rather tackle that in follow-up work, that's also fine.

**Review by songy23 — APPROVED** [review]

File: comp/core/hostname/remotehostnameimpl/hostname.go (hunk 1 above)
Comment (dineshg13) [reply comp/core/hostname/remotehostnameimpl/hostname.go:28]: I would not to add configuration for this. Customers should worry about it.  I don't think it currently configurable in any agent

**Pull request merged**